### PR TITLE
manifest: update nrf to upmerge 2023.04.17

### PR DESCRIPTION
**Pull request by ktaborowski**
[KRKNWK-16741]
https://github.com/nrfconnect/sdk-nrf/pull/10876

### CODE REVIEW

File: tests/functional/critical_region/src/main.c [hunk 1]
```diff
@@ -55,7 +55,9 @@ void test_sid_pal_critical_region_with_timer(void)
  */
 extern int unity_main(void);
 
-void main(void)
+int main(void)
 {
 	(void)unity_main();
+
+	return 0;
```
Comment (RobertGalatNordic) [inline tests/functional/critical_region/src/main.c:62]: Shouldn't we return the return value from `unity_main()`

**Review by RobertGalatNordic — APPROVED** [review]
LGTM, only minor comment

File: tests/functional/critical_region/src/main.c (hunk 1 above)
Comment (ktaborowski) [reply tests/functional/critical_region/src/main.c]: Done

**Comment by ktaborowski** [comment]
The power conumption issue needs a fix from ncs https://github.com/nrfconnect/sdk-nrf/pull/10930

**Pull request merged**